src/service/utils/setup.js

```diff
@@ -105,7 +105,13 @@ for (const path of [Config.CACHEDIR, Config.CONFIGDIR, Config.RUNTIMEDIR])
  */
 globalThis.HAVE_REMOTEINPUT = GLib.getenv('GDMSESSION') !== 'ubuntu-wayland';
 globalThis.HAVE_WAYLAND = GLib.getenv('XDG_SESSION_TYPE') === 'wayland';
-globalThis.HAVE_GNOME = GLib.getenv('GNOME_SETUP_DISPLAY') !== null;
+globalThis.HAVE_GNOME = GLib.getenv('GSCONNECT_MODE') !== 'cli';
+try {
+    // eslint-disable-next-line no-unused-expressions
+    imports.gi.Shell;
```

Comment:
Like `imports.ui`, this will always fail because `imports.gi.Shell` is a private library only available in the `gnome-shell` process.

I'd suggest using the env variable [`XDG_SESSION_DESKTOP`](https://www.freedesktop.org/software/systemd/man/latest/pam_systemd.html#Environment) and checking for [`GNOME`](https://specifications.freedesktop.org/menu-spec/latest/apb.html). This should be exported for all GNOME users:

```sh
$ echo $XDG_CURRENT_DESKTOP 
GNOME
```

Reply:
In this [Arch Wiki page](https://wiki.archlinux.org/title/environment_variables), it is said that `XDG_SESSION_DESKTOP`  is not standardized by freedesktop.org, with a link to the issue: https://gitlab.gnome.org/GNOME/gtk/-/issues/1224#note_270915